src/docs/cookbook/animation/physics-simulation.md

```diff
@@ -0,0 +1,393 @@
+`onPanEnd` callbacks. To adjust the alignment, use a `MediaQuery` to get the
+size of the widget, and divide by 2. (This converts units of "pixels dragged" to
+coordinates that
+[Align](https://api.flutter.dev/flutter/widgets/Align-class.html) uses.) Then,
+set the `Align` widget's `alignment` to `_dragAlignment`:
+
+```dart
+@override
+Widget build(BuildContext context) {
+  var size = MediaQuery.of(context).size;
+  return GestureDetector(
+    onPanDown: (details) {},
+    onPanUpdate: (details) {
+      setState(() {
+        _dragAlignment += Alignment(
+          details.delta.dx / (size.width / 2),
+          details.delta.dy / (size.height / 2),
+        );
+      });
+    },
+    onPanEnd: (details) {},
+    child: Align(
+      child: Card(
+        child: widget.child,
+      ),
+    ),
+  );
+}
+```
+
+## Step 3: Animate the widget
+
+When the widget is released, it should spring back to the center.
+
+Add an `Animation<Alignment>` field and an `_updateAnimation()` method. This
+method defines a `Tween` that interpolates between the point the widget was
+dragged to, to the point in the center .
+
+```dart
+Animation<Alignment> _animation;
+
+void _updateAnimation() {
+ _animation = _controller.drive(
+   AlignmentTween(
+     begin: _dragAlignment,
+     end: Alignment.center,
+   ),
+ );
+}
+```
+
+Call `_updateAnimation` in `initState`. This gives `_animation` an initial
+value. Then update `_dragAlignment` when the `AnimationController` produces a
+value:
+
+```dart
+@override
+void initState() {
+  _controller = AnimationController(vsync: this, duration: Duration(seconds: 1));
+  _controller.addListener(() {
+    setState(() {
+      _dragAlignment = _animation.value;
+    });
+  });
+
+  _updateAnimation();
+  super.initState();
+}
+
+```
+
+Next, make the `Align` widget use the `_dragAlignment` field:
+
+```dart
+child: Align(
+  alignment: _dragAlignment,
+  child: Card(
+    child: widget.child,
+  ),
+),
+```
+
+
+Finally, update the `GestureDetector` to manage the animation controller:
+
+
+```dart
+onPanDown: (details) {
+ _controller.stop();
+},
+onPanUpdate: (details) {
+ setState(() {
+   _dragAlignment += Alignment(
+     details.delta.dx / (size.width / 2),
+     details.delta.dy / (size.height / 2),
+   );
+ });
+},
+onPanEnd: (details) {
+ _updateAnimation();
+ _controller.reset();
+ _controller.forward();
+},
+```
+
+## Step 4: Calculate the velocity to simulate a springing motion
+
+The last step is to do a little math, to calculate the velocity of the widget
+after it's finished being dragged. This is so that the widget realistically
+continues at that speed before being snapped back. ( `_updateAnimation()`
+already sets the direction  by setting the animation's start and end alignment.)
+
+The `onPanEnd` callback provides a `DragEndDetails` object with the same
+velocity that the user was dragging their finger at. This value is in pixels per
```

Comment:
Normally, I'm not so picky, but I can't have this sentence end with a preposition.  :)
=> ... object with the same velocity as the user's finger drag.

Also removes "their" and the whole gender issue.  Not that I love this sentence either, but I think it's less confusing.

Reply:
Yeah this sentence is a bit problematic... I've reworked it to try to match the docs more closely:

```
The `onPanEnd` callback provides a [DragEndDetails][] object. This object
provides the velocity of the pointer when it stopped contacting the screen. The
velocity is in pixels per second, but the `Align` widget doesn't use pixels. It
uses coordinate
```